docs/motivation.md

```diff
@@ -0,0 +1,553 @@
+subtyping, classes are treated nominally, and traits are treated structurally.
+Furthermore there is no subtyping between classes. This means a class
+can be a subtype of a trait, but never a subtype of a class other than itself.
+For example, given the definitions
+```verona
+class C {
+    equal(self: C, other: C): Bool {...}
+}
+
+class D {
+    equal(self: D, other: D): Bool {...}
+}
+
+type Eq = {
+    equal(self: Self, other: Self) : Bool
+}
+
+```
+both `C` and `D` are subtypes of `Eq`, but neither `C` or `D` are subtypes of
+the other.
+
+#### Inheritance
+There are two forms of inheritance in
+Verona. One between classes and one between class and trait.
+
+##### Class inheritance
+Inheritance between classes works as you would expect in a language like
+Java, with the exception that it does not imply subtyping. As an example,
+consider
+```verona
+class C {
+    equal(self: C, other: C): Bool {...}
+    f(...) : ... {...}
+}
+
+// here we reuse functionality of C in D, specifically the method f, but
+// we do not get a subtyping relation D <: C
+class D : C {
+    equal(self: D, other: D): Bool {...}
+    g(...): ... { ... f(...) ... }
+}
+// TODO: find a realistic example
+```
+If we want to describe a relation between `C` and `D`, we can only do so by
+subtyping them under a trait.
+
+There is a case to be made for not allowing inheritance from a
+  class, since this allows unanticipated code reuse. This might however also be
+  a positive.
+
+##### Trait inheritance
+Traits can be used for inheritance with their default methods.
+```verona
+type ToString = {
+    toString(s: Self) : String
+    print(s: Self) : Unit {
+        sys.print(s.toString())
+    }
+}
+
+class C : ToString {
+    toString(s: C) : String {
+        ...
+    }
+}
+// here C will inherit the default implementation of print
+```
+Here, the definition `ToString` contains a default implementation for the print
+method.
+
+* Question: Should traits be allowed to define default fields?
+
+###### External trait inheritance
+A possibility is to allow externally declared trait inheritance, i.e.
+declaration of trait inheritance can be separated from class definition.
+```verona
+// file 1
+
+// class C definition
+class C {
+    toString(s: C) : String
+    ...
+}
+
+// file 2
+// declaration that C implements ToString together with its default print method
+class C : ToString
+```
+
+
+#### Self types
+In Verona, the Self type represents the concrete type of which the object is
+part. When defining class types, Self mereley acts as an alias for the
+instantiation of this class type itself:
+```verona
+class C {
+    m1(self: Self) : Self
+    m2(self: C) : C
+    // in verona, these two method signatures are equivalent up to method name.
+}
+```
+
+For trait types, Self acts as as a reference to the underlying concrete type of the instantiation. This allows us to write e.g.
+```verona
+type Collection[T] = {
```

Comment:
Have you explained the `[T]` notation so far? This seemed to come out of nowhere.